pkg/auth/providers/goauth/goauth_client.go

```diff
@@ -0,0 +1,62 @@
+package goauth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+)
+
+//GClient implements a httpclient for google oauth
+type GClient struct {
+	httpClient *http.Client
+}
+
+func (g *GClient) getUser(accessToken string, config *v3.GoogleOauthConfig) (Account, error) {
+	// userinfo endpoint doesn't require viewType param, non-admins and admins both can query this endpoint
+	respBytes, _, err := g.getFromGoogle(accessToken, config.UserInfoEndpoint)
+	if err != nil {
+		return Account{}, err
+	}
+
+	var goauthAccount Account
+	if err = json.Unmarshal(respBytes, &goauthAccount); err != nil {
+		return Account{}, err
+	}
+	return goauthAccount, nil
+}
+
+func (g *GClient) getFromGoogle(accessToken string, url string) ([]byte, int, error) {
+	var statusCode int
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, statusCode, err
+	}
+	req.Header.Add("Authorization", "Bearer "+accessToken)
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("user-agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36)")
+	resp, err := g.httpClient.Do(req)
+	if err != nil {
+		logrus.Errorf("Received error from google api: %v", err)
+		return nil, statusCode, err
+	}
+	defer resp.Body.Close()
```

Comment:
lets see if this code is common between github and google oauth, and move it to the utils if it's shared.

Reply:
Some parts of this method are common, but there are differences in headers, and how github client calculates next page link (not valid for Google). Also github client file has other methods along with getFromGithub like postToGithub and I think it makes sense to let them be in the same file. I can put the headers in a common file though